src/domain_tests/TestRunner.cs

```diff
@@ -0,0 +1,1103 @@
+// We can't refer to or use Python.Runtime here.
+// We want it to be loaded only inside the subdomains
+using System;
+using Microsoft.CSharp;
+using System.CodeDom.Compiler;
+using System.IO;
+using System.Linq;
+
+namespace Python.DomainReloadTests
+{
+    /// <summary>
+    /// This class provides an executable that can run domain reload tests.
+    /// The setup is a bit complicated:
+    /// 1. pytest runs test_*.py in this directory.
+    /// 2. test_classname runs Python.DomainReloadTests.exe (this class) with an argument
+    /// 3. This class at runtime creates a directory that has both C# and
+    ///    python code, and compiles the C#.
+    /// 4. This class then runs the C# code.
+    ///
+    /// But wait there's more indirection. The C# code that's run -- known as
+    /// the test runner --
+    /// This class compiles a DLL that contains the class which code will change
+    /// and a runner executable that will run Python code referencing the class.
+    /// Each test case:
+    /// * Compiles some code, loads it into a domain, runs python that refers to it.
+    /// * Unload the domain.
+    /// * Compile a new piece of code, load it into a domain, run a new piece of python that accesses the code.
+    /// * Unload the domain. Reload the domain, run the same python again.
+    /// This class gets built into an executable which takes one argument:
+    /// which test case to run. That's because pytest assumes we'll run
+    /// everything in one process, but we really want a clean process on each
+    /// test case to test the init/reload/teardown parts of the domain reload
+    /// code.
```

Comment:
The header comment here looks like a merge error?

Reply:
Git history says it isn't. It does need some editing though

Reply:
Probably a merge error between two parts of my brain then.